src/application/components/Explorer/Explorer.tsx

```diff
@@ -64,31 +61,38 @@ function executeOperation({
       fetchPolicy,
     });
 
-    sendExplorerRequest(payload);
+    panelWindow.send({ type: "explorerRequest", payload });
 
-    listenForResponse(
-      (response) => {
-        observer.next(response);
-        if (isSubscription) {
-          const checkForSubscriptionTermination = (event: MessageEvent) => {
-            if (event.data.name.startsWith(EXPLORER_SUBSCRIPTION_TERMINATION)) {
-              sendSubscriptionTerminationRequest();
-              observer.complete();
-              window.removeEventListener(
-                "message",
-                checkForSubscriptionTermination
-              );
-            }
-          };
+    const removeListener = panelWindow.on("explorerResponse", (message) => {
+      const { payload } = message;
 
-          window.addEventListener("message", checkForSubscriptionTermination);
-        } else {
-          observer.complete();
-        }
-      },
-      operationName,
-      !!isSubscription
-    );
+      if (payload.operationName !== operationName) {
```

Comment:
This seems prone to collisions - might it make sense to add a `uid` to the `explorerRequest` and `explorerResponse`, or even add a small `panelWindow.rpc` abstraction?

Reply:
Chatted in our 1:1. For transparency, here is what we discussed:

I absolutely agree with this and think this is a valid change. This PR is mostly focused on a 1:1 port of existing code over to `actor` from `Relay`, so we will do this in a followup PR.